src/util-privs.c

```diff
@@ -153,6 +153,11 @@ int SCGetUserID(const char *user_name, const char *group_name, uint32_t *uid, ui
     uint32_t groupid = 0;
     struct passwd *pw;
 
+    if (user_name == NULL || strlen(user_name) == 0) {
+        FatalError("no user name was provided - ensure it is specified either in the configuration "
+                   "file or in command-line arguments");
```

Comment:
would be good to list the actual commandline option

Reply:
> FatalError != assert in my thinking

I guess ok. I liked FatalError because you can specify the message before exiting. 

OK, would you prefer to change only the FatalError to SCLogError and then do the assert or do the assert directly without any additional message?

Change inside of the if statement:

```
- FatalError("no user name was provided - ensure it is specified either in the configuration "
                   "file or in command-line arguments");
+ SCLogError("no user name was provided - ensure it is specified either in the configuration "
                   "file or in command-line arguments");
+ assert(user_name == NULL || strlen(user_name) == 0) // or it can contain just true as we know it is already true I guess?
```

Remove the if statement altogether and just replace it with:
```
assert(user_name == NULL || strlen(user_name) == 0)
```

Thanks.

Reply:
my point wasnt to change the code, but that the commit speaks of "assert" when FatalError is used. "assert" has a specific meaning when dealing with errors, and FatalError != assert

Reply:
right, now I got it, thanks for the clarification!